packages/flame/lib/src/components/component_set.dart

```diff
@@ -18,6 +19,19 @@ import '../../game.dart';
 /// This wrapper also guaranteed that [Component.prepare], [Loadable.onLoad]
 /// and all the lifecycle methods are called properly.
 class ComponentSet extends QueryableOrderedSet<Component> {
+  /// With default settings, creates a [ComponentSet] with the compare function
+  /// that uses the Component's priority for sorting.
+  ComponentSet({
+    int Function(Component e1, Component e2)? comparator,
+    bool? strictMode,
+  }) : super(
+          comparator: comparator ?? Comparing.on<Component>((c) => c.priority),
+          strictMode: strictMode ?? defaultStrictMode,
+        );
+
+  // When we switch to Dart 2.15 this can be replaced with constructor tear-off
+  static ComponentSet createDefault() => ComponentSet();
```

Comment:
Is this needed/used anywhere? The reason to have it before was bc it was actually configuring the created set. Now you could just call ComponentSet() directly?

Reply:
Yes, this is used by `Component.childrenFactory`. Somehow using a simple function comprehension `() => ComponentSet()` produces an analyzer warning there.

Reply:
Apparently, they fixed the warning just 8 days ago: https://github.com/dart-lang/linter/pull/3118, but it will probably take time until we can actually use that.